### PR TITLE
[onert] Introduce xnnpack backend

### PR DESCRIPTION
**Pull request by YongseopKim**
Introduce xnnpack backend

ONE-DCO-1.0-Signed-off-by: Yongseop Kim <yons.kim@samsung.com>

### CODE REVIEW

**Comment by YongseopKim** [comment]
for #4968

**Comment by YongseopKim** [comment]
xnnpack on onert: https://github.com/Samsung/ONE/pull/5034

**Comment by YongseopKim** [comment]
Current status

```
odroid@odroid:/home/dragon/Works/github/ONE$ ONERT_LOG_ENABLE=0 BACKENDS="xnnpack" OP_BACKEND_Conv2D="xnnpack" ./Product/out/bin/nnapi_test ./tests/scripts/framework/cache/conv_2d/convolution2/convolution_test2.tflite
...
[NNAPI TEST #1] PASSED
...
[NNAPI TEST #2] PASSED
```

```
odroid@odroid:/home/dragon/Works/github/ONE$ BACKENDS="cpu;xnnpack" OP_BACKEND_Conv2D="xnnpack" ./Product/out/bin/nnapi_test ./tests/scripts/framework/cache/MODELS/inception_module/inception_test.tflite
...
[NNAPI TEST #1] PASSED
...
[NNAPI TEST #2] PASSED
```

**Comment by chunseoklee** [comment]
@YongseopKim Current PR uses 1 thread ?

**Comment by YongseopKim** [comment]
@chunseoklee maybe yes. I don't touch any thread setting. Therefore, the default thread num would be set.

File: runtime/onert/backend/xnnpack/ops/DepthwiseConvolutionLayer.cc [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      _dilation_height_factor, _dilation_width_factor,
+      input_channels /* groups */,
+      1 /* group_input_channels */,
+      _multiplier /* group_output_channels */,
+      input_channels /* input_channel_stride */,
+      output_channels /* output_channel_stride */,
+      reinterpret_cast<const float*>(_kernel->buffer()),
+      reinterpret_cast<const float*>(_bias->buffer()),
+      output_activation_min, output_activation_max,
+      0 /* flags */, &_kernel_op);
```
Comment (periannath) [inline runtime/onert/backend/xnnpack/ops/DepthwiseConvolutionLayer.cc:128]: @YongseopKim `XNN_FLAG_DEPTHWISE_CONVOLUTION` flag might be needed at here. https://github.com/google/XNNPACK/blob/9cef5eabcb5cc408db468abc0009dc424a852272/src/runtime.c#L354

**Comment by YongseopKim** [comment]
```
ONERT_LOG_ENABLE=1 BACKENDS="cpu;xnnpack" OP_BACKEND_Conv2D="xnnpack" OP_BACKEND_DepthwiseConv2D="xnnpack" ./Product/out/bin/nnapi_test tests/scripts/framework/cache/depthwise_conv_2d/depthconv1/depth_conv_test1.tflite
```

**Comment by YongseopKim** [comment]
@nnfw-bot help

**Comment by YongseopKim** [comment]
@nnfw-bot test all

**Comment by YongseopKim** [comment]
TODO
- Remove permutation
- Release unused const tensors

**Comment by YongseopKim** [comment]
PTAL @periannath and @chunseoklee

**Comment by YongseopKim** [comment]
@nnfw-bot test x64-builder

**Review by chunseoklee — DISMISSED** [review]

**Comment by ragmani** [comment]
@nnfw-bot test cross-runtime-test-debu

**Comment by ragmani** [comment]
@nnfw-bot test cross-runtime-test-debug

**Comment by ragmani** [comment]
@nnfw-bot test android-runtime-builder

**Comment by YongseopKim** [comment]
Thanks @ragmani

**Comment by YongseopKim** [comment]
PTAL @chunseoklee and @periannath

File: tests/nnfw_api/src/one_op_tests/Conv2D.cc [hunk 2]
```diff
@@ -35,7 +35,7 @@ TEST_F(GenModelTest, OneOp_Conv2D)
   _context->addTestCase(uniformTCD<float>(
       {{4, 0, -5, 1, 0, 4, -1, 1, -1, -3, 3, -2, -4, 1, -2, 2, 4, -4, 2, 2, 0, 4, -1, -2, 4}},
       {{47, -4, -25, 9, 10, 10, -13, 11, -14, -26, -12, 26, 20, 40, 1, 3, 11, 4}}));
-  _context->setBackends({"acl_cl", "acl_neon", "cpu", "ruy"});
+  _context->setBackends({"acl_cl", "acl_neon", "cpu", "ruy", "xnnpack"});
```
Comment (periannath) [inline tests/nnfw_api/src/one_op_tests/Conv2D.cc:38]: I just found that you have to add `xnnpack` to `GenModelTestContext::setBackends`. https://github.com/Samsung/ONE/blob/c46ddc04abdb58323fbd38389e6927f003bfaea1/tests/nnfw_api/src/GenModelTest.h#L215

Without that line, `xnnpack` will not be tested on `GenModelTest`.

**Review by periannath — CHANGES_REQUESTED** [review]

**Comment by YongseopKim** [comment]
PTAL @chunseoklee and @periannath

**Review by periannath — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**